homeassistant/helpers/event.py

```diff
@@ -301,9 +313,10 @@ def _async_track_state_change_event(
         TRACK_STATE_CHANGE_CALLBACKS,
         TRACK_STATE_CHANGE_LISTENER,
         EVENT_STATE_CHANGED,
-        _async_dispatch_entity_id_event,
-        _async_state_change_filter,
-        action,
+        # Remove type ignores when _async_track_event uses EventType
+        _async_dispatch_entity_id_event,  # type: ignore[arg-type]
+        _async_state_change_filter,  # type: ignore[arg-type]
+        action,  # type: ignore[arg-type]
```

Comment:
I think I like this idea. It would be nice to see this filled out so we can get a better grasp on how it would impact the downstream code. I don't think this will have a performance impact since the typing isn't in the inner function but it would be nice to run it though the profiler as well to make sure it doesn't cause a performance regression.

Reply:
> I think I like this idea. It would be nice to see this filled out so we can get a better grasp on how it would impact the downstream code.

Take a look at https://github.com/home-assistant/core/compare/dev...cdce8p:ha-core:event-generic for a more complete example. I didn't include it here to keep the changes to a minimum.